tests/NHSD.BuyingCatalogue.Ordering.Common.UnitTests/Builders/OrderBuilder.cs

```diff
@@ -192,51 +191,58 @@ public OrderBuilder WithServiceInstanceId(OrderItem forOrderItem, int increment)
-            order.SupplierId = supplierId;
-            order.SupplierName = supplierName;
-            order.SupplierAddress = supplierAddress;
-            order.SupplierContact = supplierContact;
-            order.CommencementDate = commencementDate;
+            var order = new Order
+            {
+                CommencementDate = commencementDate,
+                Description = orderDescription,
+                FundingSourceOnlyGms = fundingSourceOnlyGms,
+                OrderingParty = new OrderingParty
+                {
+                    Id = organisationId,
+                    Name = organisationName,
+                    OdsCode = organisationOdsCode,
+                    Address = organisationAddress,
+                },
+                OrderingPartyContact = organisationContact,
+                Progress = new OrderProgress
+                {
+                    AdditionalServicesViewed = additionalServicesViewed,
+                    AssociatedServicesViewed = associatedServicesViewed,
+                    CatalogueSolutionsViewed = catalogueSolutionsViewed,
+                    ServiceRecipientsViewed = serviceRecipientsViewed,
+                },
+                Supplier = new Supplier
+                {
+                    Id = supplierId,
+                    Name = supplierName,
+                    Address = supplierAddress,
+                },
+                SupplierContact = supplierContact,
+            };
 
             foreach (var orderItem in orderItems)
             {
-                order.AddOrderItem(orderItem, lastUpdatedBy, lastUpdatedByName);
+                order.AddOrUpdateOrderItem(orderItem);
             }
 
-            order.SetServiceRecipients(
-                serviceRecipients,
-                lastUpdatedBy,
-                lastUpdatedByName);
-
-            order.AdditionalServicesViewed = additionalServicesViewed;
-            order.ServiceRecipientsViewed = serviceRecipientsViewed;
-            order.CatalogueSolutionsViewed = catalogueSolutionsViewed;
-            order.AssociatedServicesViewed = associatedServicesViewed;
-            order.FundingSourceOnlyGms = fundingSourceOnlyGms;
-            order.Created = created;
-            order.LastUpdated = lastUpdated;
+            order.SetSelectedServiceRecipients(serviceRecipients);
 
-            Field.Set(order, nameof(Order.ServiceInstanceItems), serviceInstanceItems);
+            BackingField.SetValue(order, nameof(Order.LastUpdated), lastUpdated);
+            BackingField.SetValue(order, nameof(Order.Id), orderId);
+            BackingField.SetValue(order, nameof(Order.ServiceInstanceItems), serviceInstanceItems);
 
             if (completed is not null)
             {
-                Field.Set(order, nameof(Order.Completed), completed);
+                BackingField.SetValue(order, nameof(Order.Completed), completed);
             }
 
             return order;
```

Comment:
There's probably quite a but of functionality that can be removed from this class but we can pick that up as tech debt. Hopefully we'll be able to remove this class eventually (and rely solely on AutoFixture).